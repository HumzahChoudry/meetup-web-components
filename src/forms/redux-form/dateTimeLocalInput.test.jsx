import React from 'react';
import { shallow } from 'enzyme';
import ReduxFormDateTimeLocalInput from './DateTimeLocalInput';

describe('redux-form DateTimeLocalInput', () => {

	// props structured like redux form will pass them
	const fieldProps = {
		input: {
			label: 'Y2K',
			name: 'y2k_date',
			required: true,
			value:' 2000-01-01T00:01',
			min: '1999-12-31T23:55',
			max: '2017-06-30T16:30',
		},
		meta: {
			error: 'Too soon.'
		}
	};

	it('renders a DateTimeLocalInput with expected props from mock data', () => {
		const component = shallow(<ReduxFormDateTimeLocalInput {...fieldProps} />);
		expect(component).toMatchSnapshot();
	});

});
