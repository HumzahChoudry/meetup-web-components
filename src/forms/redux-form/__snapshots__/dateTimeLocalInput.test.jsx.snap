// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`redux-form DateTimeLocalInput renders a DateTimeLocalInput with expected props from mock data 1`] = `
<WithErrorList(DateTimeLocalInput)
  error="Too soon."
  label="Y2K"
  max="2017-06-30T16:30"
  min="1999-12-31T23:55"
  name="y2k_date"
  required={true}
  value=" 2000-01-01T00:01"
/>
`;
